### PR TITLE
Refactor SDK managed workload install

### PR DESCRIPTION
**Pull request by sfoslund**
cba1d1d addresses leftover PR feedback from https://github.com/dotnet/sdk/pull/16530

### CODE REVIEW

**Comment by wli3** [comment]
@dsplaisted @sfoslund why we started to spin more projects? it means more dlls to deal with. We decided in 3.0 time not to create more dlls.

**Comment by wli3** [comment]
everything else looks good

**Comment by sfoslund** [comment]
> why we started to spin more projects?

@dsplaisted mentioned that the workload resolver will have to reference the IAL, so I separated workload installation into its own project that can be referenced from the CLI or resolver. Workload install should be compiled into the resolver dll.

**Comment by wli3** [comment]
@dsplaisted reference workload install will pull entire nuget dependency and bunch of nuget dlls in. That won't work.

**Comment by dsplaisted** [comment]
> @dsplaisted reference workload install will pull entire nuget dependency and bunch of nuget dlls in. That won't work.

The workload resolver needs to at least be able to read the workload installation records so that it can see which workloads are installed.  So it looks like we may need to split the IAL into two different projects.

**Comment by sfoslund** [comment]
@dsplaisted @wli3 I separated out the workload installation record library methods and included them in the workload resolver as we discussed, let me know if you have feedback.

File: src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallRecords/SdkFeatureBand.cs [hunk 1]
```diff
@@ -4,7 +4,8 @@
 using System;
 using Microsoft.Deployment.DotNet.Releases;
 
-namespace Microsoft.DotNet.Workloads.Workload.Install
+#nullable disable
+namespace Microsoft.DotNet.Workloads.Workload.Install.InstallRecord
```
Comment (wli3) [inline src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallRecords/SdkFeatureBand.cs:8]: Sorry I missed that in the beginning. The following method looks wrong. Release version is just a Version right? isn't comparison should be only the major, minor and build % 100.

There is also no validation. An SDK version band cannot be 5.0.1 or 5.0 only 5.0.1xx
public bool Equals(SdkFeatureBand other)
        {
            return ToString() == other.ToString();
        }
        public int CompareTo(SdkFeatureBand other)
        {
            return string.Compare(ToString(), other.ToString(), StringComparison.Ordinal);
        }

**Review by wli3 — APPROVED** [review]

**Review by dsplaisted — APPROVED** [review]

**Pull request merged**